### DISCUSSION TITLE
Access to User Properties in the HTTP Access control feature

### DISCUSSION THREAD

**Discussion by estebanz01** [emqx/emqx#11528] [General]
Hola! 👋 

Just wondering if it's possible, having a `CONNECT` packet with `User-Property`s, to use that metadata in the HTTP Authorization request to an external HTTP API.

right now, my endpoint does not have the necessary information to properly validate a client as we need to use username/password and a API key that is going to be hardcoded. This client is a device so a JWT authentication won't fit.

Thanks!

**Comment by lafirest** [comment]
@estebanz01 
Yes, of course able. You can see [here](https://www.emqx.io/docs/en/v5/access-control/authn/http.html#configure-with-dashboard) how to use an external HTTP server for authentication with client information.

**Reply by estebanz01** [reply]
thanks for the reply! Maybe I wasn't clear enough 😅 or I'm not sure if I'm reading correctly, but I want something like this:

* A client sets a `User-Property` of (`ApiKey: ABC123`) in the `CONNECT` packet along with a `username` and a `password`.
* EMQx sees this and sends as a POST request a `username`, `password` and `ApiKey: ABC123` to an external HTTP service.
* The HTTP Service receives a JSON object like this: `{"username": ..., "password": ..., "apiKey": "ABC123" }`

from what I gather on the docs, there are certain placeholders like `username`, `password`, `clientid`, `peerhost` and some `cert_*` ones, but not something like `user_properties`.

Can you help me understand if what I want is possible or no and where can I find info to grab those user properties and send them to the HTTP endpoint if possible?

Thanks again!

**Reply by lafirest** [reply]
This is a good question, now the authentication is based on client information, not the connection information namely what you want, a workaround way is to concatenate your password and API key into one string and use it as the password.
And then this could be converted to a feature request issue.

**Reply by estebanz01** [reply]
I'll open an issue to track this as a feature request. Thanks!

**Reply by estebanz01** [reply]
created one: https://github.com/emqx/eip/issues/78

**Comment by estebanz01** [comment]
I posted a request feature here: https://github.com/emqx/eip/issues/78
